pkg/scheduler/plugin.go

```diff
@@ -615,13 +615,12 @@ func (k *deleter) deleteOne(pod *Pod) error {
 	}
 }
 
-// Create creates a scheduler plugin and all supporting background functions.
+// Create a scheduler plugin and all supporting background functions.
 func (k *KubernetesScheduler) NewPluginConfig(terminate <-chan struct{}, mux *http.ServeMux) *PluginConfig {
-
 	// Watch and queue pods that need scheduling.
 	updates := make(chan queue.Entry, k.schedcfg.UpdatesBacklog)
 	podUpdates := &podStoreAdapter{queue.NewHistorical(updates)}
-	reflector := cache.NewReflector(createAllPodsLW(k.client), &api.Pod{}, podUpdates, 0)
+	reflector := cache.NewReflector(k.podsListWatch, &api.Pod{}, podUpdates, 0)
```

Comment:
Agree, that would be cleaner. Was happy to get that mocked without faking the whole apiserver watch API first.